__init__.py

```diff
@@ -106,40 +113,93 @@ def respond(self, query):
+        # for fallback but try to get the 
+        # real abstract
+        if r.type == 'disambiguation':
+            if r.related:
+                detailed_url = r.related[0].url + "?o=x"
+                LOG.debug("DDG: disambiguating %s" % (detailed_url,))
+                request = requests.get(detailed_url)
+                response = request.content
+                if response:
+                    xml = ElementTree.fromstring(response)
+                    r = ddg.Results(xml)
 
         if (r.answer is not None and r.answer.text and
                 "HASH" not in r.answer.text):
             return(query + self.is_verb + r.answer.text + '.')
         elif len(r.abstract.text) > 0:
             sents = split_sentences(r.abstract.text)
-            return sents[0]
+            #return sents[0]  # what it is
+            #return sents     # what it should be
+            return ". ".join(sents)   # what works for now
         elif len(r.related) > 0 and len(r.related[0].text) > 0:
             related = split_sentences(r.related[0].text)[0]
             return(self.format_related(related, query))
         else:
             return None
 
+    def fix_input(self, query):
+        for noun in self.translated_question_words:
+            for verb in self.translated_question_verbs:
+                for article in [i + ' ' for i in self.translated_articles] + ['']:
+                    test = noun + verb + ' ' + article
+                    if query[:len(test)] == test:
+                        return query[len(test):]
+        return query
+
     def CQS_match_query_phrase(self, query):
         answer = None
-        for noun in self.question_words:
-            for verb in self.question_verbs:
-                for article in [i + ' ' for i in self.articles] + ['']:
+        for noun in self.translated_question_words:
+            for verb in self.translated_question_verbs:
+                for article in [i + ' ' for i in self.translated_articles] + ['']:
                     test = noun + verb + ' ' + article
                     if query[:len(test)] == test:
                         answer = self.respond(query[len(test):])
                         break
         if answer:
             return (query, CQSMatchLevel.CATEGORY, answer)
         else:
+            LOG.debug("DDG has no answer")
             return None
 
     def stop(self):
         pass
 
+    @intent_handler(IntentBuilder("AskDucky").require("DuckDuckGo"))
+    def handle_ask_ducky(self, message):
+        """entry point when wiki is called out by name
+           in the utterance"""
+        utt = message.data['utterance']
+
+        if utt is None:
+            return
+
+        for voc in self.sorted_vocab:
+            utt = utt.replace(voc,"")
+
+        utt = utt.strip()
+        utt = self.fix_input(utt)
+        utt = utt.replace("an ","")   # ugh!
+        utt = utt.replace("a ","")   # ugh!
+        utt = utt.replace("the ","")   # ugh!
```

Comment:
if the request is made in a non-English language, this will fail.  could change it to loop over `self.translated_articles` instead.  Then you will only need one `# ugh!` comment ;-)

Reply:
Yea, the answer is no because of the spacing. This is an ugly hack just to get this skill propped up (see the # ugh! after each line) so I will add this to the feature request ticket that this be revisited in the future and cleaned up in some manner. Also note the nested parsing approach referenced above will also only work for English as non English sentences are not structured in this manner. These sorts of corrections will require some time if someone wishes to get involved at that level. See Jira ticket skill-242 for further follow up.